dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModelManager.java

```diff
@@ -0,0 +1,61 @@
+package com.dat3m.dartagnan.verification.model;
+
+import com.dat3m.dartagnan.encoding.EncodingContext;
+import com.dat3m.dartagnan.program.memory.MemoryObject;
+import com.dat3m.dartagnan.verification.model.event.EventModelManager;
+import com.dat3m.dartagnan.verification.model.relation.RelationModelManager;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.Formula;
+import org.sosy_lab.java_smt.api.Model;
+
+import java.math.BigInteger;
+import java.util.List;
+
+
+public class ExecutionModelManager {
+    private final EventModelManager eMManager;
+    private final RelationModelManager rMManager;
+
+    private Model model;
+
+    public ExecutionModelManager(){
+        eMManager = new EventModelManager(this);
+        rMManager = new RelationModelManager(this);
+    }
+
+    public ExecutionModelNext buildExecutionModel(EncodingContext context, Model model, boolean buildAsConfig)
+        throws InvalidConfigurationException
+    {
+        this.model = model;
+        final ExecutionModelNext executionModel = new ExecutionModelNext();
+
+        eMManager.buildEventModels(executionModel, context);
+        extractMemoryLayout(executionModel, context);
+        rMManager.buildRelationModels(executionModel, context, buildAsConfig);
+
+        this.model = null;
+
+        return executionModel;
+    }
```

Comment:
As far as I can see, you are passing the `ExecutionModelManager` to each of `EventModelManager` and `RelationModelManager` just so you can alter get access to the model. Why not simply add the model as an argument to `ExecutionModelManager`, pass them to the two "subManagers" and methods like `manager.evaluateByModel` or `manager.isTrue`

Reply:
I agree that this doesn't feel nice. I'm wondering if the explicit decomposition into `EventModelManager` and `RelationModelManager` is event needed.

Reply:
For the functionality it is OK to have only one manager to do all things instead of three. But I think the reason why I designed like this is to have an identical structure for everything related to `ExecutionModelManagerNext`, `EventModel` and `RelationModel`, i.e. to have the same structure not only for the data but also for their managers.

Reply:
But is that structure necessary on the manager level? As a user, I would never interact with any of those manager classes except calling a single `getExecutionModelFromSMTModel(Model smtModel)` function, no?
With the division into separate classes, you suddenly need to expose a lot of internal functionality that a user should never use, just so the other managers can access it.